DigiMeSDK/Core/Classes/Security/DMEJWTManager.swift

```diff
@@ -0,0 +1,444 @@
+            case state
+            case timestamp
+        }
+    }
+    
+    // claims to validate pre- authorization code
+    class PayloadValidatePreauthJWT: NSObject, Decodable, Claims {
+        var preauthorizationCode: String?
+
+        enum CodingKeys: String, CodingKey {
+            case preauthorizationCode = "preauthorization_code"
+        }
+    }
+    
+    // claims to request a authorization code
+    class PayloadRequestAuthJWT: NSObject, Decodable, Claims {
+        var clientId: String?
+        var code: String?
+        var codeVerifier: String?
+        var grantType: String?
+        var nonce: String?
+        var redirectUrl: String?
+        var timestamp: Double?
+
+        enum CodingKeys: String, CodingKey {
+            case clientId = "client_id"
+            case code
+            case codeVerifier = "code_verifier"
+            case grantType = "grant_type"
+            case nonce
+            case redirectUrl = "redirect_uri"
+            case timestamp
+        }
+    }
+    
+    // claims to validate authorization and refresh tokens
+    class PayloadValidateAuthJWT: NSObject, Decodable, Claims {
+        var accessToken: String?
+        var expiresTimestamp: Double?
+        var refreshToken: String?
+        var tokenType: String?
+        
+        enum CodingKeys: String, CodingKey {
+            case accessToken = "access_token"
+            case expiresTimestamp = "expires_on"
+            case refreshToken = "refresh_token"
+            case tokenType = "token_type"
+        }
+    }
+    
+    // claims to request data trigger
+    class PayloadDataTriggerJWT: NSObject, Decodable, Claims {
+        var accessToken: String?
+        var clientId: String?
+        var nonce: String?
+        var redirectUrl: String?
+        var sessionKey: String?
+        var timestamp: Double?
+
+        enum CodingKeys: String, CodingKey {
+            case accessToken = "access_token"
+            case clientId = "client_id"
+            case nonce
+            case redirectUrl = "redirect_uri"
+            case sessionKey = "session_key"
+            case timestamp
+        }
+    }
+    
+    // claims to request OAuth token renewal
+    class PayloadRefreshOAuthJWT: NSObject, Decodable, Claims {
+        var clientId: String?
+        var grantType: String?
+        var nonce: String?
+        var redirectUrl: String?
+        var refreshToken: String?
+        var timestamp: Double?
+        
+        enum CodingKeys: String, CodingKey {
+            case clientId = "client_id"
+            case grantType = "grant_type"
+            case nonce
+            case redirectUrl = "redirect_uri"
+            case refreshToken = "refresh_token"
+            case timestamp
+        }
+    }
+    
+    public override init() {
+        super.init()
+        
+        // header we use everytime we create our JWT. The second `alg` parameter will be created automatically by this library, we can't define it manually
```

Comment:
Please also update the comment, perhaps something reserved like:

`// Default JWT header`